apps/api/src/app/inbox/usecases/get-topic-subscription/get-topic-subscription.usecase.ts

```diff
@@ -38,22 +48,135 @@ export class GetTopicSubscription {
       type: PreferencesTypeEnum.SUBSCRIPTION_SUBSCRIBER_WORKFLOW,
     });
 
-    const preferencesWorkflowIds = preferencesEntities
-      .map((pref) => pref._templateId?.toString())
-      .filter((id): id is string => id !== undefined);
+    const { allPreferencesEntities, allWorkflowEntities } = await this.resolveWorkflowPreferences(
+      command,
+      subscription,
+      preferencesEntities
+    );
+
+    return mapTopicSubscriptionToDto(subscription, allPreferencesEntities, allWorkflowEntities);
+  }
+
+  private async resolveWorkflowPreferences(
+    command: GetTopicSubscriptionCommand,
+    subscription: TopicSubscribersEntity,
+    storedPreferences: PreferencesEntity[]
+  ): Promise<{
+    allPreferencesEntities: PreferencesEntity[];
+    allWorkflowEntities: SelectedWorkflowFields[];
+  }> {
+    const storedPreferenceWorkflowInternalIds = new Set(
+      storedPreferences.map((pref) => pref._templateId?.toString()).filter((id): id is string => id !== undefined)
+    );
 
     const workflowEntities =
-      preferencesWorkflowIds.length > 0
+      storedPreferenceWorkflowInternalIds.size > 0
         ? await this.notificationTemplateRepository.find(
             {
-              _id: { $in: preferencesWorkflowIds },
+              _id: { $in: Array.from(storedPreferenceWorkflowInternalIds) },
               _environmentId: subscription._environmentId,
               _organizationId: subscription._organizationId,
             },
             SELECTED_WORKFLOW_FIELDS_PROJECTION
           )
         : [];
 
-    return mapTopicSubscriptionToDto(subscription, preferencesEntities, workflowEntities);
+    const storedWorkflowIds = workflowEntities.map((workflow) => workflow.triggers?.[0]?.identifier);
+    const requestedWorkflows = await this.fetchRequestedWorkflows(command, subscription, storedWorkflowIds);
```

Comment:
it seems that this query to db can be combined with the above `await this.notificationTemplateRepository.find`, so we just do one mongo call